File: adapter/src/main/java/com/github/lany192/adapter/OneToManyFlow.java
package com.github.lany192.adapter;

import androidx.annotation.CheckResult;
import androidx.annotation.NonNull;

/**
 * Process and flow operators for one-to-many.
 */
public interface OneToManyFlow<T> {

    /**
     * Sets some item view binders to the item type.
     *
     * @param binders the item view binders
     * @return end flow operator
     */
    @CheckResult
    @SuppressWarnings("unchecked")
    @NonNull
    OneToManyEndpoint<T> to(@NonNull ItemViewBinder<T, ?>... binders);
}

File: library/src/main/java/com/github/lany192/box/multitype/OneToManyFlow.kt


package com.github.lany192.box.multitype

import androidx.annotation.CheckResult

/**
 * Process and flow operators for one-to-many.
 *
 * @author Drakeet Xu
 */
interface OneToManyFlow<T> {

  /**
   * Sets some item view delegates to the item type.
   *
   * @param delegates the item view delegates
   * @return end flow operator
   */
  @CheckResult
  fun to(vararg delegates: ItemViewDelegate<T, *>): OneToManyEndpoint<T>

  @CheckResult
  fun to(vararg delegates: ItemViewBinder<T, *>): OneToManyEndpoint<T>
}
